src/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -361,432 +363,383 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
   );
 
-  return (
-    <div>
-      <Formik
-        initialValues={initialValues}
-        validationSchema={SendAssetsSchema}
-        onSubmit={(fields) => {
-          onComplete(fields);
-        }}
-        children={({ errors, setFieldValue, setFieldTouched, touched, values }) => {
-          useEffect(() => {
-            if (updateFormValues) {
-              updateFormValues(values);
-            }
-          }, [values]);
+  const formik = useFormik({
+    initialValues,
+    validationSchema: SendAssetsSchema,
+    onSubmit: (fields) => {
+      onComplete(fields);
+    }
+  });
+
+  const { values, setFieldValue, setFieldTouched, setFieldError, errors, touched } = formik;
+
+  useEffect(() => {
+    if (updateFormValues) {
+      updateFormValues(values);
+    }
+  }, [values]);
+
+  useEffect(() => {
+    handleNonceEstimate(values.account);
+  }, [values.account]);
+
+  // Set gas estimates if default asset is selected
+  useEffectOnce(() => {
+    if (!isEmpty(values.asset)) {
+      fetchGasPriceEstimates(values.network).then((data) => {
+        setFieldValue('gasEstimates', data);
+        setFieldValue('gasPriceSlider', data.fast);
+      });
+    }
+  });
+
+  useEffect(() => {
+    if (ptxState.protectTxShow) {
+      if (goToInitialStepOrFetchReport && ptxState.receiverAddress !== values.address.value) {
+        const { address, network } = values;
+        goToInitialStepOrFetchReport(address.value, network);
+      }
+    }
+  }, [values.address.value]);
+
+  const toggleAdvancedOptions = () => {
+    setFieldValue('advancedTransaction', !values.advancedTransaction);
+  };
+  const toggleIsAutoGasSet = () => {
+    // save value because setFieldValue method is async and values are not yet updated	            // save value because setFieldValue method is async and values are not yet updated
+    const isEnablingAutoGas = !values.isAutoGasSet;
+    setFieldValue('isAutoGasSet', !values.isAutoGasSet);
+    if (isEnablingAutoGas) {
+      handleGasEstimate(true);
+    }
+  };
+  const handleGasEstimate = async (forceEstimate: boolean = false) => {
+    if (
+      values &&
+      values.network &&
+      values.asset &&
+      values.address &&
+      isValidETHAddress(values.address.value) &&
+      values.account &&
+      isValidPositiveNumber(values.amount) &&
+      (values.isAutoGasSet || forceEstimate)
+    ) {
+      setIsEstimatingGasLimit(true);
+      const finalTx = processFormForEstimateGas(values);
+      const gas = await getGasEstimate(values.network, finalTx);
+      setFieldValue('gasLimitField', hexToNumber(gas).toString());
+      setFieldTouched('amount');
+      setIsEstimatingGasLimit(false);
+    } else {
+      return;
+    }
+  };
+
+  const handleFieldReset = () => {
+    const resetFields: (keyof IFormikFields)[] = [
+      'account',
+      'amount',
+      'txDataField',
+      'advancedTransaction'
+    ];
+    resetFields.forEach((field) => setFieldValue(field, initialValues[field]));
+  };
```

Comment:
I think we can just use `handleReset` here.
https://formik.org/docs/api/formik#handlereset---void
It would save use 3 rerenders.

Reply:
I dont think so, because we don't want to reset all fields. Just these 4.

Reply:
nonce should change with the account.
So should gasEstimates when changing asset.
Doing a single `handleReset({...initialValues, asset: values.asset })` seems more efficient then making 4+2 calls.